src/resolvers/message.ts

```diff
@@ -40,55 +40,28 @@ const resolver: Resolvers = {
         }
 
         const authUsers = [...new Set([...users, auth.userId])];
-        const channels = await channelModel.findAll({
-          group: ['channelId'],
-          having: sequelize.where(
-            sequelize.fn('COUNT', sequelize.col('channelId')),
-            { [Op.in]: [authUsers.length] },
-          ),
-          include: [
-            {
-              model: membershipModel,
-              as: 'memberships',
-              where: {
-                userId: authUsers,
-              },
-              attributes: [
-                'channelId',
-              ],
-            },
-          ],
-          raw: true,
-        });
-
-        let retrievedChannelId: string;
-        const channelPromises = [];
-
-        channels.forEach((channel, index) => {
-          channelPromises[index++] = channelModel.findOne({
-            attributes: [[sequelize.fn('COUNT', sequelize.col('channelId')), 'numOfMemberships']],
-            where: {
-              id: channel.id,
-            },
-            group: ['channelId'],
-            include: [
-              {
-                model: membershipModel,
-                as: 'memberships',
-                attributes: [
-                  'channelId',
-                ],
-              },
-            ],
-            raw: true,
-          }).then((foundChannel) => {
-            if (foundChannel.numOfMemberships === authUsers.length) {
-              retrievedChannelId = channel.id;
-            }
-          });
-        });
-
-        await Promise.all(channelPromises);
+        const channels: Array<Channel> = await sequelize.query(`
+            SELECT DISTINCT c.id, COUNT(c.id) FROM channels c
+            INNER JOIN memberships m ON c.id = m.channelId
+              AND m.deletedAt IS NULL
+              AND m.userId IN (:authUsers)
+              AND (
+                SELECT COUNT(m2.channelId)
+                FROM channels c2
+                INNER JOIN memberships m2 ON c2.id = m2.channelId
+                WHERE c2.id = c.id
+                GROUP BY m2.channelId
```

Comment:
It looks like n^4 performance draining. I feel like this case significantly shows that we need a group key. How do you think about it?

Reply:
@hyochan 
Ok. I agree with your opinion of making a group key. 
I missed looking at plan of the above sql query and didn't know it has such bad performance.
I'm wondering how you found out it has n^4 performance.
Did you happen to check 'Explain Execution Plan' in DBeaver? And read the number of same rows in the 'Rows' column except the one under the nested_loop?

If so, does the query below has n^2 performance?  This query is just for a question.
```
SELECT c.id FROM channels c
INNER JOIN memberships m ON c.id = m.channelId
  AND m.deletedAt IS NULL
  AND m.userId IN
        ('9a59f370-5ba1-11ea-baed-e927eab06e55', 'c4cfe890-5fc4-11ea-8c3d-c9155d1819cc')
  AND (
    SELECT COUNT(m2.channelId)
    FROM channels c2
    INNER JOIN memberships m2 ON c2.id = m2.channelId
    WHERE c2.id = c.id
    GROUP BY m2.channelId
  ) = 2
WHERE c.deletedAt IS NULL
```